File: 203_RemoveLinkedListElements.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
//27.45%
struct ListNode* removeElements(struct ListNode* head, int val) {
    while(head && head -> val == val)
		head = head -> next;
	struct ListNode *pre = head;
	while(pre){
		while(pre -> next && pre -> next -> val == val){
			pre -> next = pre -> next -> next;
		}
		pre = pre -> next;
	}
	return head;
}

File: 203_RemoveLinkedListElements.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None
# 29.83%
class Solution(object):
    def removeElements(self, head, val):
        """
        :type head: ListNode
        :type val: int
        :rtype: ListNode
        """
        while head and head.val == val: head = head.next
        pre = head
        while pre:
            while pre.next and pre.next.val == val:
                pre.next = pre.next.next
            pre = pre.next
        return head